### ISSUE TITLE
memory error

### ISSUE THREAD

**Issue by Liron-2** [matterport/Mask_RCNN#174]
Hi,

When I run trainning, with 10 steps per epoch, and max_gt_instances = 100 ,on 2 GPUs,
I get Memory error. Do you have an idea where is the problem? 
If max_gt_instances = 20, it is working. 

(I have 1 class and more than 100 objects in one image from the same class. My masks are numpy arrays from the shape (height, width , no of instances) )
 
Thank you,
Liron

File "/home/liron/.local/lib/python3.5/site-packages/keras/utils/data_utils.py", line 636, in data_generator_task
    self.queue.put((True, generator_output))
  File "<string>", line 2, in put
  File "/usr/lib/python3.5/multiprocessing/managers.py", line 716, in _callmethod
    conn.send((self._id, methodname, args, kwds))
  File "/usr/lib/python3.5/multiprocessing/connection.py", line 206, in send
    self._send_bytes(ForkingPickler.dumps(obj))
  File "/usr/lib/python3.5/multiprocessing/reduction.py", line 50, in dumps
    cls(buf, protocol).dump(obj)
**MemoryError**
  
File "my_data.py", line 344, in <module>
    layers='heads')
  File "Mask_RCNN-master/model.py", line 2095, in train
    **fit_kwargs
  File "/home/liron/.local/lib/python3.5/site-packages/keras/legacy/interfaces.py", line 87, in wrapper
    return func(*args, **kwargs)
  File "/home/liron/.local/lib/python3.5/site-packages/keras/engine/training.py", line 2115, in fit_generator
    generator_output = next(output_generator)
  File "/home/liron/.local/lib/python3.5/site-packages/keras/utils/data_utils.py", line 735, in get
    six.reraise(value.__class__, value, value.__traceback__)
  File "/usr/local/lib/python3.5/dist-packages/six.py", line 693, in reraise
    raise value
**MemoryError**

**Comment by Pelups** [comment]
How much memory your GPUs have ?
Maybe you need to reduce the batch size.

**Comment by Liron-2** [comment]
Each of them has 11 GB memory.
The batch size is 2. (2 GPUs * 1 image )

**Comment by YanWang2014** [comment]
Hi Liron, can you share your config file? I also have more than 100 objects, in this case, how to set the following parameters:
`
RPN_TRAIN_ANCHORS_PER_IMAGE
TRAIN_ROIS_PER_IMAGE
MAX_GT_INSTANCES = 100
DETECTION_MAX_INSTANCES
`

I mean, if I set `MAX_GT_INSTANCES = 100`, how should I set the other three?

**Comment by nicholasadr** [comment]
@Liron-2 Did you manage to fix it? If so, what did you do?
